File: editor/src/de/fatox/meta/ui/tabs/ProjectHomeTab.java
package de.fatox.meta.ui.tabs;

import com.badlogic.gdx.scenes.scene2d.ui.Table;
import com.badlogic.gdx.utils.Align;
import com.kotcrab.vis.ui.widget.VisLabel;
import com.kotcrab.vis.ui.widget.VisTable;
import de.fatox.meta.api.dao.MetaProjectData;
import de.fatox.meta.injection.Inject;
import de.fatox.meta.ui.MetaEditorUI;
import de.fatox.meta.ui.components.TextWidget;
import de.fatox.meta.ui.windows.AssetDiscovererWindow;
import de.fatox.meta.ui.windows.ShaderComposerWindow;
import de.fatox.meta.ui.windows.ShaderLibraryWindow;

/**
 * Created by Frotty on 06.06.2016.
 */
public class ProjectHomeTab extends MetaTab {
    private final MetaProjectData projectData;
    private VisTable visTable = new VisTable();

    @Inject
    private MetaEditorUI editorUI;

    public ProjectHomeTab(MetaProjectData metaProjectData) {
        super(true, false);
        this.projectData = metaProjectData;
        setupTable();
    }

    private void setupTable() {
        visTable.top();
        visTable.row().height(128);
        visTable.add(new TextWidget(projectData.name));
        visTable.row().height(64);
        visTable.add();
        visTable.row();
        VisLabel visLabel = new VisLabel("This is your project home tab.");
        visLabel.setAlignment(Align.center);
        visTable.add(visLabel).padBottom(128);
    }


    @Override
    public String getTabTitle() {
        return "home@" + projectData.name;
    }

    @Override
    public Table getContentTable() {
        return visTable;
    }

    @Override
    public void onShow() {
        super.onShow();
        editorUI.metaToolbar.clear();
        editorUI.metaToolbar.addAvailableWindow(AssetDiscovererWindow.class, null);
        editorUI.metaToolbar.addAvailableWindow(ShaderLibraryWindow.class, null);
        editorUI.metaToolbar.addAvailableWindow(ShaderComposerWindow.class, null);
    }
}

File: editor/src/main/kotlin/de/fatox/meta/ui/tabs/ProjectHomeTab.kt
package de.fatox.meta.ui.tabs

import com.badlogic.gdx.scenes.scene2d.ui.Table
import com.badlogic.gdx.utils.Align
import de.fatox.meta.api.model.MetaProjectData
import de.fatox.meta.injection.MetaInject.Companion.lazyInject
import de.fatox.meta.ui.MetaEditorUI
import de.fatox.meta.ui.components.MetaLabel
import de.fatox.meta.ui.components.MetaTable
import de.fatox.meta.ui.components.TextWidget
import de.fatox.meta.ui.windows.AssetDiscovererWindow
import de.fatox.meta.ui.windows.ShaderComposerWindow
import de.fatox.meta.ui.windows.ShaderLibraryWindow

/**
 * Created by Frotty on 06.06.2016.
 */
class ProjectHomeTab(private val projectData: MetaProjectData) : MetaTab(true, false) {
	private val visTable = MetaTable()

	private val editorUI: MetaEditorUI by lazyInject()

	override fun getTabTitle(): String {
		return "home@" + projectData.name
	}

	override fun getContentTable(): Table {
		return visTable
	}

	override fun onShow() {
		super.onShow()
		editorUI.metaToolbar.apply {
			clear()
			addAvailableWindow(AssetDiscovererWindow::class, null)
			addAvailableWindow(ShaderLibraryWindow::class, null)
			addAvailableWindow(ShaderComposerWindow::class, null)
		}
	}

	init {
		val visLabel = MetaLabel("This is your project home tab.", 16).apply { setAlignment(Align.center) }

		visTable.apply {
			top()
			row().height(128f)
			add(TextWidget(projectData.name))
			row().height(64f)
			add()
			row()
			add(visLabel).padBottom(128f)
		}
	}
}
